### PR TITLE
Add unity test e2e automation

### PR DESCRIPTION
**Pull request by JacobGros**
# Description
This PR adds automation for secret and storageclass creation for the Unity E2E test, updating the README to include pre-reqs that were left out, and fixes yaml lint errors 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1449 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran Unity e2e test, verified it passed 
No sc/secret on cluster:
```
# kubectl get secret -n unity
No resources found in unity namespace.
# kubectl get sc
NAME        PROVISIONER             RECLAIMPOLICY   VOLUMEBINDINGMODE      ALLOWVOLUMEEXPANSION   AGE
unity       csi-unity.dellemc.com   Delete          WaitForFirstConsumer   true                   2d23h
unity-nfs   csi-unity.dellemc.com   Delete          WaitForFirstConsumer   true                   2d23h 
```
Then we run Unity test:
```
# ./run-e2e-test.sh --unity --no-modules
...
STEP: Starting: Install Unity Driver(Standalone)  @ 09/09/24 12:23:26.48
  STEP:      Executing  Given an environment with k8s or openshift, and CSM operator installed @ 09/09/24 12:23:26.48
  STEP:      Executing  Create storageclass with name [op-e2e-unity] and template [testfiles/unity-templates/unity-storageclass-template.yaml] for [unity] @ 09/09/24 12:23:26.49
  STEP:      Executing  Set up secret with template [testfiles/unity-templates/unity-secret-template.yaml] name [unity-creds] in namespace [unity] for [unity] @ 09/09/24 12:23:26.896
  STEP:      Executing  Apply custom resource [1] @ 09/09/24 12:23:27.217
  Sep  9 12:23:27.217: INFO: Running '/usr/bin/kubectl --namespace=unity apply --validate=true -f -'
  Sep  9 12:23:27.349: INFO: stderr: ""
  Sep  9 12:23:27.349: INFO: stdout: "containerstoragemodule.storage.dell.com/unity created\n"
  STEP:      Executing  Validate custom resource [1] @ 09/09/24 12:23:27.349
  STEP:      Executing  Validate [unity] driver from CR [1] is installed @ 09/09/24 12:24:27.354
  STEP:      Executing  Run custom test @ 09/09/24 12:25:27.36
  Sep  9 12:25:27.360: INFO: Running cert-csi [test vio --sc op-e2e-unity --chainNumber 2 --chainLength 2]
[2024-09-09 12:25:27]  INFO Starting cert-csi; ver. 1.5.0
[2024-09-09 12:25:27]  INFO Using EVENT observer type
[2024-09-09 12:25:27]  INFO Using config from /root/.kube/config
[2024-09-09 12:25:27]  INFO Successfully loaded config. Host: https://x.x.x.x
[2024-09-09 12:25:27]  INFO Created new KubeClient
[2024-09-09 12:25:27]  INFO Running 1 iteration(s)
[2024-09-09 12:25:27]  INFO     *** ITERATION NUMBER 1 ***
[2024-09-09 12:25:27]  INFO Starting VolumeIoSuite with op-e2e-unity storage class
[2024-09-09 12:25:27]  INFO Successfully created namespace volumeio-test-8d1f82b4
[2024-09-09 12:25:27]  INFO Using default number of volumes
[2024-09-09 12:25:27]  INFO Using default image: docker.io/centos:latest
[2024-09-09 12:25:27]  INFO Creating IO pod
[2024-09-09 12:25:27]  INFO Waiting for pod iowriter-test-q4tzs to be READY
[2024-09-09 12:25:27]  INFO Waiting for pod iowriter-test-bb22l to be READY
[2024-09-09 12:26:03]  INFO Executing command: [/bin/bash -c dd if=/dev/urandom bs=1M count=128 oflag=sync > /data0/writer-0.data]
[2024-09-09 12:26:03]  INFO Executing command: [/bin/bash -c dd if=/dev/urandom bs=1M count=128 oflag=sync > /data0/writer-1.data]
[2024-09-09 12:26:09]  INFO Executing command: [/bin/bash -c sha512sum /data0/writer-0.data > /data0/writer-0.sha512]
[2024-09-09 12:26:10]  INFO Executing command: [/bin/bash -c sha512sum /data0/writer-1.data > /data0/writer-1.sha512]
[2024-09-09 12:26:12]  INFO Waiting until no Volume Attachments with PV  left
[2024-09-09 12:26:12]  INFO VolumeAttachment deleted
[2024-09-09 12:26:12]  INFO Waiting for pod iowriter-test-6kq7c to be READY
[2024-09-09 12:26:14]  INFO Waiting until no Volume Attachments with PV  left
[2024-09-09 12:26:14]  INFO VolumeAttachment deleted
[2024-09-09 12:26:14]  INFO Waiting for pod iowriter-test-mm92z to be READY
[2024-09-09 12:26:18]  INFO Executing command: [/bin/bash -c sha512sum -c /data0/writer-0.sha512]
[2024-09-09 12:26:22]  INFO Executing command: [/bin/bash -c sha512sum -c /data0/writer-1.sha512]
[2024-09-09 12:26:24]  INFO Hashes match
[2024-09-09 12:26:24]  INFO Executing command: [/bin/bash -c dd if=/dev/urandom bs=1M count=128 oflag=sync > /data0/writer-0.data]
[2024-09-09 12:26:28]  INFO Executing command: [/bin/bash -c sha512sum /data0/writer-0.data > /data0/writer-0.sha512]
[2024-09-09 12:26:30]  INFO Waiting until no Volume Attachments with PV  left
[2024-09-09 12:26:30]  INFO VolumeAttachment deleted
[2024-09-09 12:26:31]  INFO Hashes match
[2024-09-09 12:26:31]  INFO Executing command: [/bin/bash -c dd if=/dev/urandom bs=1M count=128 oflag=sync > /data0/writer-1.data]
[2024-09-09 12:26:35]  INFO Executing command: [/bin/bash -c sha512sum /data0/writer-1.data > /data0/writer-1.sha512]
[2024-09-09 12:26:40]  INFO Waiting until no Volume Attachments with PV  left
[2024-09-09 12:26:40]  INFO VolumeAttachment deleted
[2024-09-09 12:26:40]  INFO Deleting all resources in namespace volumeio-test-8d1f82b4
[2024-09-09 12:26:52]  INFO Namespace volumeio-test-8d1f82b4 was deleted in 12.018994036s
[2024-09-09 12:26:52]  INFO Waiting for volumeattachments to be deleted
[2024-09-09 12:26:54]  INFO SUCCESS: VolumeIoSuite in 1m27.117769971s
[2024-09-09 12:26:54]  INFO Started generating reports...
Collecting metrics
1 / 1 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s[2024-09-09 12:26:54]  INFO Started generating reports...
Collecting metrics
Generating plots
1 / 1 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s1 / 1 [-----------------------------------------------------------------------------------------------------------------------------------------------------------------------] 100.00% ? p/s[2024-09-09 12:26:54]  WARN No ResourceUsageMetrics provided
[2024-09-09 12:26:54] ERROR no ResourceUsageMetrics provided
report-test-run-898e2c85:
Name: test-run-898e2c85
Host: https://x.x.x.x
StorageClass: op-e2e-unity
Minimum and Maximum EntityOverTime charts:/root/.cert-csi/reports/test-run-898e2c85/PodsCreatingOverTime.png/root/.cert-csi/reports/test-run-898e2c85/PodsReadyOverTime.png/root/.cert-csi/reports/test-run-898e2c85/PodsTerminatingOverTime.png/root/.cert-csi/reports/test-run-898e2c85/PvcsCreatingOverTime.png/root/.cert-csi/reports/test-run-898e2c85/PvcsBoundOverTime.pngTests:
--------------------------------------------------------------
1. TestCase: VolumeIoSuite
            Started:   2024-09-09 12:25:27.425093965 -0500 -0500
            Ended:     2024-09-09 12:26:54.542853771 -0500 -0500
            Result:    SUCCESS            Stage metrics:
                    PVCAttachment:
                        Avg: 6.472096466s
                        Min: 6.261408365s
                        Max: 6.682784568s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCAttachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCAttachment_boxplot.png
                    PVCBind:
                        Avg: 23.887488714s
                        Min: 22.834408947s
                        Max: 24.940568481s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCBind.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCBind_boxplot.png
                    PVCCreation:
                        Avg: 30.572367076s
                        Min: 29.780383897s
                        Max: 31.364350255s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCCreation_boxplot.png
                    PVCDeletion:
                        Avg: 150.868353ms
                        Min: 115.47502ms
                        Max: 186.261687ms
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCDeletion_boxplot.png
                    PVCUnattachment:
                        Avg: 5.726516485s
                        Min: 5.574599899s
                        Max: 5.878433072s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCUnattachment.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PVCUnattachment_boxplot.png
                    PodCreation:
                        Avg: 21.037024773s
                        Min: 5.039170153s
                        Max: 36.048282335s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PodCreation.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PodCreation_boxplot.png
                    PodDeletion:
                        Avg: 1.951267906s
                        Min: 1.645368564s
                        Max: 2.160794844s
                        Histogram:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PodDeletion.png
                        BoxPlot:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/PodDeletion_boxplot.png
                        EntityNumberOverTime:
        /root/.cert-csi/reports/test-run-898e2c85/VolumeIoSuite1/EntityNumberOverTime.png[2024-09-09 12:26:54]  INFO Avg time of a run:   72.89s
[2024-09-09 12:26:54]  INFO Avg time of a del:   12.02s
[2024-09-09 12:26:54]  INFO Avg time of all:     87.11s
[2024-09-09 12:26:54]  INFO During this run 100.0% of suites succeeded
  STEP:      Executing  Enable forceRemoveDriver on CR [1] @ 09/09/24 12:26:54.952
  STEP:      Executing  Delete custom resource [1] @ 09/09/24 12:26:54.975
  STEP:      Executing  Restore template [testfiles/unity-templates/unity-secret-template.yaml] for [unity] @ 09/09/24 12:26:54.997
  STEP:      Executing  Restore template [testfiles/unity-templates/unity-storageclass-template.yaml] for [unity] @ 09/09/24 12:26:55.093
  STEP: Ending: Install Unity Driver(Standalone)
 
```
- [x] Installed PowerFlex CSM to make sure other yaml lint errors were ok

### CODE REVIEW

**Review by alexemc — APPROVED** [review]

File: tests/README.md [hunk 1]
```diff
@@ -51,7 +58,7 @@ go get github.com/onsi/gomega/...
 
 ### Array Information
 
-For PowerFlex, PowerScale, Authorization, and Application-Mobility, system-specific information (array login credentials, system IDs, endpoints, etc.) need to be provided so that all the required resources (secrets, storageclasses, etc.) can be created by the tests. Example values have been inserted; please replace these with values from your system. Refer to [CSM documentation](https://dell.github.io/csm-docs/docs/) for any further questions about driver or module pre-requisites.
+For PowerFlex, Unity, PowerScale, Authorization, and Application-Mobility, system-specific information (array login credentials, system IDs, endpoints, etc.) need to be provided so that all the required resources (secrets, storageclasses, etc.) can be created by the tests. Example values have been inserted; please replace these with values from your system. Refer to [CSM documentation](https://dell.github.io/csm-docs/docs/) for any further questions about driver or module pre-requisites.
```
Comment (jooseppi-luna) [inline tests/README.md:61]: Should PowerStore now be included here as well?

**Review by jooseppi-luna — CHANGES_REQUESTED** [review]
Just one tiny request

File: tests/README.md (hunk 1 above)
Comment (JacobGros) [reply tests/README.md:61]: we have a separate subtask for powerstore, will likely be opening another PR to update docs for that as we work on it

**Review by jooseppi-luna — APPROVED** [review]
kek

**Pull request merged**